app/concerns/long_runnable.rb

```diff
@@ -65,6 +66,7 @@ def run
 
     def run!
       @thread = Thread.new do
+        Thread.current[:name] = "#{id}-#{Time.now}"
```

Comment:
I was going for the human readable format because I used rbtrace to get information about the running and orphaned threads, not sure if we would ever use that programmatically.